graph/temporal.go

```diff
@@ -0,0 +1,17 @@
+package graph
+
+type TemporalLine interface {
```

Comment:
This type needs thought (and a doc comment).

From what I can see, `At` and `Duration` (more below) are always called together, so they could be merged to a single method, `Interval() (start, end T)`(?) reducing the interface method dispatch cost.

`Duration` is suggestive that it would return a `time.Duration`. So a different name would be better even without the above change. Alternatively, `T` above is made a `time.Duration` or `time.Time`.